api/main_endpoints/models/DoorCode.js

```diff
@@ -5,6 +5,7 @@ const DoorCodeSchema = new Schema(
   {
     doorCode: {
```

Comment:
we should add a check to ensure the doorcode follows a format of `xxx-xxxx`.

Check out mongoose's `pre('save')` function ([here](https://mongoosejs.com/docs/middleware.html#pre)) to add a check that the string is correct. an example where we use the `pre` hook is in [`api/main_endpoints/models/User.js`](https://github.com/SCE-Development/Core-v4/blob/be6648edd785a45e9fd2eac2037b3218faca9ffd/api/main_endpoints/models/User.js#L70)

check 2 things:
1. the format has three characters, a dash, then 4 characters
1. all characters beides the dash are numbers

i highly reccomend using regular expressions for this ([example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test))